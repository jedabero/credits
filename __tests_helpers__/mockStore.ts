// eslint-disable-next-line import/no-extraneous-dependencies
import configureMockStore from 'redux-mock-store';
import { RootState } from '../src/store';

export default configureMockStore<RootState>();
