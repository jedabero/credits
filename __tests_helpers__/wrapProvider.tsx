import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import { Store } from '../src/store';

const wrapProvider = (component: ReactElement, store: Store) => (
  <Provider store={store}>{component}</Provider>
);

export default wrapProvider;
