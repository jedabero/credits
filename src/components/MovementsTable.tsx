import { EntityId } from '@reduxjs/toolkit';
import { selectMovementsIdsByCardId } from '../reducers/movements';
import { useAppSelector } from '../hooks';
import MovementRow from './MovementRow';

type MovementsTableProps = {
  cardId: EntityId,
  onEdit: React.Dispatch<React.SetStateAction<EntityId | undefined>>,
  onView: React.Dispatch<React.SetStateAction<EntityId | undefined>>,
};

function MovementsTable({ cardId, onEdit, onView }: MovementsTableProps) {
  const movementsIds = useAppSelector(selectMovementsIdsByCardId(cardId));

  return (
    <div className="movementstable">
      <div className="movementrow">
        <div>ID</div>
        <div>Auth Number</div>
        <div>Date</div>
        <div>Description</div>
        <div>Value</div>
        <div>Monthly Rate</div>
        <div>Yearly Rate</div>
        <div>Quotas</div>
        <div>Actions</div>
      </div>
      {movementsIds.map((id) => (
        <MovementRow key={id} movementId={id} onEdit={onEdit} onView={onView} />
      ))}
    </div>
  );
}

export default MovementsTable;
