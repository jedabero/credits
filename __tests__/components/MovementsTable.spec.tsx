/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { create } from 'react-test-renderer';
import MovementsTable from '../../src/components/MovementsTable';
import wrapProvider from '../../__tests_helpers__/wrapProvider';
import mockStore from '../../__tests_helpers__/mockStore';

describe('MovementsTable test', () => {
  test('should render', () => {
    const store = mockStore({
      _persist: { version: 0, rehydrated: true },
      creditCards: {
        entities: {},
        ids: [],
      },
      movements: {
        entities: {},
        ids: [],
      },
    });
    const tree = create(
      wrapProvider(
        <MovementsTable cardId="SOME ID" onEdit={() => {}} onView={() => {}} />,
        store,
      ),
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
