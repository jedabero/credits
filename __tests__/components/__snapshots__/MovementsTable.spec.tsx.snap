// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovementsTable test should render 1`] = `
<div
  className="movementstable"
>
  <div
    className="movementrow"
  >
    <div>
      ID
    </div>
    <div>
      Auth Number
    </div>
    <div>
      Date
    </div>
    <div>
      Description
    </div>
    <div>
      Value
    </div>
    <div>
      Monthly Rate
    </div>
    <div>
      Yearly Rate
    </div>
    <div>
      Quotas
    </div>
    <div>
      Actions
    </div>
  </div>
</div>
`;
